src/algorithms/DominguezRios.jl

```diff
@@ -0,0 +1,267 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    DominguezRios()
+
+`DominguezRios` implements the algorithm of Dominguez-Rios et al. (2021).
+"""
+
+mutable struct DominguezRios <: AbstractAlgorithm end
+
+mutable struct _Box
+    l::Vector{Float64}
+    u::Vector{Float64}
+    priority::Float64
+    function _Box(l::Vector{Float64}, u::Vector{Float64}, priority::Float64)
+        @assert length(l) == length(u) "Dimension mismatch between l and u"
+        return new(l, u, priority)
+    end
+end
+
+_Box(l::Vector{Float64}, u::Vector{Float64}) = _Box(l, u, 0.0)
+
+function _scaled_priority(
+    l::Vector{Float64},
+    u::Vector{Float64},
+    yI::Vector{Float64},
+    yN::Vector{Float64},
+)
+    return prod((u - l) ./ (yN - yI))
+end
+
+function _scaled_priority(B::_Box, yI::Vector{Float64}, yN::Vector{Float64})
+    return _scaled_priority(B.l, B.u, yI, yN)
+end
+
+function _reduced_scaled_priority(
+    l::Vector{Float64},
+    u::Vector{Float64},
+    i::Int,
+    z::Vector{Float64},
+    yI::Vector{Float64},
+    yN::Vector{Float64},
+)
+    if i ≠ length(z)
```

Comment:
```suggestion
    if I != length(z)
```

This is that unicode thing again. I know it looks cool, but it's easier in the long-term if we just use mostly ASCII

Reply:
Sorry about that. I just copied some parts from my implementation and since it worked I skipped checking it I guess.